File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/category/dto/CategoryDto.java
package com.side.tiggle.domain.category.dto;

import com.side.tiggle.domain.category.model.Category;
import com.side.tiggle.domain.category.model.CategoryType;
import lombok.*;
import lombok.experimental.SuperBuilder;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@SuperBuilder
public class CategoryDto {

    private String name;
    private CategoryType type;
    private Boolean defaults;

    public static CategoryDto fromEntity(Category category) {
        return CategoryDto.builder()
                .name(category.getName())
                .type(category.getType())
                .defaults(category.isDefaults())
                .build();
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/category/dto/CategoryDto.kt
package com.side.tiggle.domain.category.dto

import com.side.tiggle.domain.category.model.Category
import com.side.tiggle.domain.member.model.Member

data class CategoryDto(
    val name: String,
    val defaults: Boolean,
    val memberId: Long
) {

    fun toEntity(
        member: Member
    ): Category {
        return Category(
            member = member,
            name = name,
            defaults = defaults
        )
    }
    companion object {
        fun fromEntity(category: Category): CategoryDto {
            return CategoryDto(
                memberId = category.member.id,
                name = category.name,
                defaults = category.defaults
            )
        }
    }
}